primitives/runtime/src/traits.rs

```diff
@@ -1260,11 +1260,11 @@ pub trait AccountIdConversion<AccountId>: Sized {
 
 /// Format is TYPE_ID ++ encode(parachain ID) ++ 00.... where 00... is indefinite trailing zeroes to
 /// fill AccountId.
-impl<T: Encode + Decode + Default, Id: Encode + Decode + TypeId> AccountIdConversion<T> for Id {
+impl<T: Encode + Decode, Id: Encode + Decode + TypeId> AccountIdConversion<T> for Id {
 	fn into_sub_account<S: Encode>(&self, sub: S) -> T {
 		(Id::TYPE_ID, self, sub)
 			.using_encoded(|b| T::decode(&mut TrailingZeroInput(b)))
-			.unwrap_or_default()
+			.expect("`AccountId` type is never greater than 32 bytes; qed")
```

Comment:
Doesn't this and all the other "type is never greater than 32bytes" defeats the entire point of having the `AccountId` generic? 

I mean this is right for Polkadot, but someone could come up with whatever account id they want?

Reply:
Well, on Moonbeam it's 20-bytes, so I don't think it defeats the point. In any case, it's actually fine here since there are trailing zeroes.

Reply:
I thought about this again. Don't we maybe want to create a custom trait for generating an account id? Instead of abusing `Decode` for this job.

Reply:
Yeah that would make sense as a further refactor. But I wouldn't want to hold this up for that given the level of bitrot it would suffer by even 7 days sitting around.